src/Dumplie/Infrastructure/Doctrine/Metadata/DoctrineStorage.php

```diff
@@ -0,0 +1,238 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Dumplie\Infrastructure\Doctrine\Dbal\Metadata;
+
+use Doctrine\DBAL\Connection;
+use Dumplie\Application\Metadata\Schema;
+use Dumplie\Application\Metadata\Storage;
+
+class DoctrineStorage implements Storage
+{
+    const TABLE_PREFIX = 'metadata';
+
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    /**
+     * @var TypeMapping[]
+     */
+    private $mapping;
+
+    /**
+     * DoctrineStorage constructor.
+     *
+     * @param Connection $connection
+     * @param array      $typeMappings
+     */
+    public function __construct(Connection $connection, array $typeMappings = [])
+    {
+        $this->connection = $connection;
+        foreach ($typeMappings as $mapping) {
+            $this->addMapping($mapping);
+        }
+    }
+
+    /**
+     * @param Schema $schema
+     *
+     * @throws DoctrineStorageException
+     */
+    public function create(Schema $schema)
+    {
+        $schemaManager = $this->connection->getSchemaManager();
+        $schemaAsset = $schemaManager->createSchema();
+
+        foreach ($schema->types() as $type) {
+            $tableName = $this->tableName($schema->name(), $type->name());
+
+            if ($schemaManager->tablesExist([$tableName])) {
+                throw DoctrineStorageException::tableAlreadyExists($tableName);
+            }
+
+            $table = $schemaAsset->createTable($tableName);
+
+            $table->addColumn('id', 'string');
+            $table->setPrimaryKey(['id']);
+
+            foreach ($type->getDefinitions(['id']) as $field => $definition) {
+                if (!array_key_exists($definition->type(), $this->mapping)) {
+                    throw DoctrineStorageException::unableToMapField($definition->type());
+                }
+
+                $this->mapping[$definition->name()]->map($table, $field, $definition);
+            }
+        }
+
+        $queries = $schemaAsset->toSql($this->connection->getDatabasePlatform());
+        $this->executeQueries($queries);
```

Comment:
I would not. Some engines have implicit commit when you try to define/modify db objects (eg. in MySQL).